interview-questions.md

```diff
@@ -8,36 +8,243 @@ Then, research the question to expand on your answer. Even if you feel you have
+
+Researched answer: 
+    I ran these different commands now and checked to see what they actually did and not just a guess... OH MY GOD! There is a css file created when you run the Rails generate controler (Animal). That was really the biggest thing i just learned. I also learened that you can use the generate resource command but if you forgot to add rspec you will have to add rspec and because you already created everything else without rspec you will have to add it to the different models and controllers bucasue that is what the generate resourse command would have done for you had you had it. the generate resouse command is awesome you just have to make sure you do the proper steps prior.
+
+$ rails generate model(Animal legs:string eyes:number food:string birthday:date) 
+    *stuff in () is the model, will change per project
+    Is used to create a model from my console. it does not have commas between the differnt things and dont forget the foreign key if you the model you are creating needs it, see answer 1 if you screw that up. You can find what ws created by this by going to db>schema.rb
+
+$ Rails generate controller (Animal)
+    *stuff in () is the controller name, will change per project
+    Is used to create a controller file (app>contorllers>animal_controller.rb) and the view folder (app>views>main), but not a view file that is important. 
+    The controller is the brains, it takes in the RESTful routes and decides what methods do what it is very very important. 
+    The other thing created when the generate controller command is run is the View folder. If you want to pair your controller and model and route, assuming you created a route, you will have to view it somehow. 
+    We can create a file in our view folder called animal.html.erb, the erb means embedded ruby.
+    now we can make this page look however we want
+
+
+$ rails generate resource (Animal legs:string eyes:number food:string birthday:date) 
+    *stuff in () is the model a=made
+        This command is super cool and save me a lot of work. it creates a model, creates rspec files, creates a controller, creates a a view and creates some helpers, and lastly it creates a route! this thing is so awesome. it just did all the work of the other 2 commands before it plus a lot of thinking and the making of the route and just got rid of it. But there is a downfall, i did not make it, so i am constantly checking routes-E to see what the routes are and if they have changed for whatever reason
+
+
```

Comment:
nice examples...I think you wore using the parenthesis just to give emphasis to the portion of the generate command that changes. However, we would not run the command with those parenthesis. Unless you have research that shows different.